### DISCUSSION TITLE
configuration file for dev settings

### DISCUSSION THREAD

**Discussion by LeoTuet** [resend/react-email#578] [Ideas]
After seeing a lot of issues relating to dev server configuration like #248 and #372, I think it makes sense to introduce a config file like `react-email.config.js` (similar to next having `next.config.js` or vite with `vite.config.ts`) on top of commandline args. This file could then also be typed which makes configuration more intuitive.

I would also be happy to implement it :)

**Comment by bukinoshita** [comment]
Can you share more on this and how this would work. I had some early thoughts on this when creating `react-email` but still wasn't 100% sure.

**Comment by LeoTuet** [comment]
`react-email` would expose a function called e.g `configureReactEmail` which has a typed object as a param and just return the object again.


Then you would import and default export this function into your `react-email.config.js`
```ts
import { configureReactEmail } from "react-email"

export default configureReactEmail({
	srcDir: "src"
})

```
(Its the same as just having export default {} in `react-email.config.js`. The function is just purely there to enable type hinting)


Then it is possible to create a function/variable which just imports the default export of `react-email.config.js`
```ts
const config = (await import(`${projectRoot}/react-email.config.js`)).default
```

**Comment by zenorocha** [comment]
I'd love for us to stay away from this as much as we can:
https://twitter.com/wesbos/status/1620793926306205701/photo/1
